src/wp-includes/theme.php

```diff
@@ -866,6 +866,13 @@ function switch_theme( $stylesheet ) {
 
 	update_option( 'theme_switched', $old_theme->get_stylesheet() );
 
+	// Set autoload=no for the previous theme
+	$old_theme_stylesheet = $old_theme->get_stylesheet();
+	wp_set_option_autoload( "theme_mods_$old_theme_stylesheet", 'no' );
```

Comment:
This adjustment alters the autoload option for the previous theme. If the user had activated multiple themes previously, it becomes essential to verify and modify the autoload value for those themes as well. What do you think?

Reply:
@mukeshpanchal27 I think this is only an issue for existing sites that had multiple themes active before updating to WordPress 6.5. So I think this is better addressed separately via your new ticket https://core.trac.wordpress.org/ticket/59975 and #5711.

Any site that starts with WordPress 6.5 should not have this problem as every time the theme is switched the previous theme's theme mods are set to not autoload. So there should at no point be more than one theme with autoloaded theme mods.